test/performance/latency/LatencyTestPublisher.cpp

```diff
@@ -177,409 +199,466 @@ bool LatencyTestPublisher::init(
-        {
-            // Factor of 2 below is to calculate the roundtrip divided by two. Note that the overhead does not
-            // need to be halved, as we access the clock twice per round trip
-            latency_publisher_->end_time_ = std::chrono::steady_clock::now();
-            latency_publisher_->times_.push_back(std::chrono::duration<double, std::micro>(
-                        latency_publisher_->end_time_ - latency_publisher_->start_time_) / 2. -
-                    latency_publisher_->overhead_time_);
-            latency_publisher_->received_count_++;
-
-            // Reset seqnum from out data
-            latency_publisher_->dynamic_data_type_out_->set_uint32_value(0, 0);
-
-            latency_publisher_->mutex_.lock();
-            ++latency_publisher_->data_msg_count_;
-            if (latency_publisher_->data_msg_count_ >= latency_publisher_->subscribers_)
-            {
-                latency_publisher_->data_msg_cv_.notify_one();
-            }
-            latency_publisher_->mutex_.unlock();
-        }
+        latency_publisher_->dynamic_data_type_out_->set_uint32_value(0, 0);
     }
     else
     {
-        subscriber->takeNextData((void*)latency_publisher_->latency_type_in_, &latency_publisher_->sampleinfo_);
-        if (latency_publisher_->latency_type_in_->seqnum == latency_publisher_->latency_type_out_->seqnum)
-        {
-            latency_publisher_->end_time_ = std::chrono::steady_clock::now();
-            latency_publisher_->times_.push_back(std::chrono::duration<double, std::micro>(
-                        latency_publisher_->end_time_ - latency_publisher_->start_time_) / 2. -
-                    latency_publisher_->overhead_time_);
-            latency_publisher_->received_count_++;
-
-            latency_publisher_->mutex_.lock();
-            ++latency_publisher_->data_msg_count_;
-            if (latency_publisher_->data_msg_count_ >= latency_publisher_->subscribers_)
-            {
-                // Reset seqnum from out data
-                latency_publisher_->latency_type_out_->seqnum = 0;
-                latency_publisher_->data_msg_cv_.notify_one();
-            }
-            latency_publisher_->mutex_.unlock();
-        }
+        latency_publisher_->latency_type_out_->seqnum = 0;
+    }
+
+    ++latency_publisher_->data_msg_count_;
+    if (latency_publisher_->data_msg_count_ >= latency_publisher_->subscribers_)
+    {
+        latency_publisher_->data_msg_cv_.notify_one();
     }
 }
 
 void LatencyTestPublisher::run()
 {
     // WAIT FOR THE DISCOVERY PROCESS FO FINISH:
     // EACH SUBSCRIBER NEEDS 4 Matchings (2 publishers and 2 subscribers)
-    std::unique_lock<std::mutex> disc_lock(mutex_);
-    while (discovery_count_ != (subscribers_ * 4))
-    {
-        discovery_cv_.wait(disc_lock);
-    }
+    unique_lock<mutex> disc_lock(mutex_);
+    discovery_cv_.wait(
+        disc_lock,
+        [this]() -> bool
+        {
+            return total_matches() == 4;
```

Comment:
There could be more than one latency subscriber running...

```suggestion
            return total_matches() == 4 * subscribers_;
```

Reply:
Done in 38d969f4b12f32e97ec60f8b44d02e591713638e